File: Data Structures & Algorithms/validate-parentheses/submission-0.java
class Solution {
    public boolean isValid(String s) {
        if(s.length()==0||s.length()%2!=0)
        return false;
        Stack <Character> stack=new Stack<>();
        char[] ch=s.toCharArray();
        for(int i=0;i<s.length();i++)
        {
            if(ch[i]=='('||ch[i]=='{'||ch[i]=='[')
            {
                stack.push(ch[i]);
            }
            else
            {
                if (stack.isEmpty()) return false;
                char op=stack.pop();
                if((ch[i]==')' && op!='(')||(ch[i]==']' && op!='[')||(ch[i]=='}' && op!='{') )
                return false;
            }
            
        }
        return stack.isEmpty();

        
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.cs
public class Solution
{
    public int EvalRPN(string[] tokens)
    {
        Stack<int> stack = new();

        foreach (string token in tokens)
        {
            if (int.TryParse(token, out int number))
            {
                stack.Push(number);
                continue;
            }

            int second = stack.Pop();
            int first = stack.Pop();

            int result = token switch
            {
                "+" => first + second,
                "-" => first - second,
                "*" => first * second,
                "/" => first / second,
                _ => throw new ArgumentException("Invalid operator")
            };

            stack.Push(result);
        }

        return stack.Pop();
    }
}